provider/cmd/helpers.go

```diff
@@ -58,3 +71,49 @@ func providerFromFlags(flags *pflag.FlagSet) (sdk.Address, error) {
 
 	return addr, nil
 }
+
+func providersFromFlagsOrQuery(flags *pflag.FlagSet, ctx context.Context, cctx client.Context, did dtypes.DeploymentID) ([]sdk.Address, error) {
+	var providers []string
+
+	if !flags.Changed(FlagProvider) {
+		cclient := akashclient.NewQueryClientFromCtx(cctx)
+		resp, err := cclient.Leases(ctx, &mtypes.QueryLeasesRequest{
+			Filters: mtypes.LeaseFilters{
+				Owner: did.Owner,
+				DSeq:  did.DSeq,
+				State: mtypes.Lease_State_name[int32(mtypes.LeaseActive)],
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		for _, lease := range resp.Leases {
+			providers = append(providers, lease.Lease.LeaseID.Provider)
```

Comment:
Could we wind up with the same provider address multiple times here?

Reply:
there shouldn't be multiple leases for one dseq on provider